code-challenges.test.js

```diff
@@ -13,61 +13,134 @@
+//input: object
+//output: [15, 15, 20, 30, 30, 60, 90], [10, 15, 20, 45, 60, 65, 100]
+// describe("reSort", () => {
+//   it("takes in an object and returns an array of the values sorted from least to greatest", () => {
+//     const studyMinutesWeek1 = {
+//       sunday: 90,
+//       monday: 30,
+//       tuesday: 20,
+//       wednesday: 15,
+//       thursday: 30,
+//       friday: 15,
+//       saturday: 60
+//     }
+//     // Expected output: [15, 15, 20, 30, 30, 60, 90]
+    
+//     const studyMinutesWeek2 = {
+//       sunday: 100,
+//       monday: 10,
+//       tuesday: 45,
+//       wednesday: 60,
+//       thursday: 20,
+//       friday: 15,
+//       saturday: 65
+//     }
+//     expect(reSort(studyMinutesWeek1)).toEqual([15, 15, 20, 30, 30, 60, 90])
+//     expect(reSort(studyMinutesWeek2)).toEqual([10, 15, 20, 45, 60, 65, 100])
+//   })
+// })
+//     // Expected output: [10, 15, 20, 45, 60, 65, 100]
+// // b) Create the function that makes the test pass.
+
+// // Pseudo code:
+// // create a new variable 
+// // that takes in object as input to extract the values from the object, sorts them using sort method .sort()
+// // use (a, b) => a - b for sorting
+// //
+// // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+
+// const reSort = (obj) => {
+//   const valuesArray = Object.values(obj)
+//   valuesArray.sort((a, b) => a - b)
+//   return valuesArray
+// }
 
 // a) Create a test with expect statements for each of the variables provided.
+//input: [100, -17, -23, -9], [250, -89, 100, -96], []
+//output: [100, 83, 60, 51], [250, 161, 261, 165], []
 
-const accountTransactions1 = [100, -17, -23, -9]
-// Expected output: [100, 83, 60, 51]
 
-const accountTransactions2 = [250, -89, 100, -96]
-// Expected output: [250, 161, 261, 165]
+describe("addMe", () => {
+it("add each number from each other", () => {
+  const accountTransactions1 = [100, -17, -23, -9]
+  // Expected output: [100, 83, 60, 51]
+  const accountTransactions2 = [250, -89, 100, -96]
+  // Expected output: [250, 161, 261, 165]
+  const accountTransactions3 = []
+  // Expected output: []
+  expect(addMe(accountTransactions1)).toEqual[100, -17, -23, -9]
+  expect(addMe(accountTransactions2)).toEqual[250, -89, 100, -96]
+  // expect(subMe(accountTransactions3)).toEqual[]
+})
+})
 
-const accountTransactions3 = []
-// Expected output: []
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+//create a function 
+// create a new varible with empty array
+// utilize a for loop
+//iterate through the input array and add each number with the next one, pushing the into a new array .push() returning our new array newTrans
+
+const addMe = (arr) => {
+  const newTrans = []
+
+  for (let i = 0; i<arr.length -1; i++) {
+    newTrans.push(arr[i] + arr[i + 1])
```

Comment:
When you correct your test, this code will not pass. The main issue is that it's adding the current value to the value at the next index, but it's not keeping track of the accumulating sum.